newscoop/src/Newscoop/NewscoopBundle/Security/Http/Authentication/FrontendLogoutSuccessHandler.php

```diff
@@ -50,4 +51,15 @@ public function onLogoutSuccess(Request $request)
 
         return parent::onLogoutSuccess($request);
     }
+
+    private function setTargetUrl(Request $request)
```

Comment:
This `everything` can be one-liner.

```
$this->targetUrl = $request->query->get('_target_path', $request->query->get('url', null));
```

Keeping two if's in separate function doesn't make it simpler to read.